src/main/java/com/byznass/tiolktrack/kernel/handler/GpsLocationById.java

```diff
@@ -0,0 +1,37 @@
+package com.byznass.tiolktrack.kernel.handler;
+
+import com.byznass.tiolktrack.kernel.dao.GpsProvider;
+import com.byznass.tiolktrack.kernel.model.Gps;
+import com.byznass.tiolktrack.kernel.model.Location;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+
+public class GpsLocationById {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(GpsLocationById.class);
+
+	private final GpsProvider gpsProvider;
+
+	@Inject
+	GpsLocationById(GpsProvider gpsProvider) {
+
+		this.gpsProvider = gpsProvider;
+	}
+
+	public Location execute(String gpsId) {
+
+		LOGGER.info("Trying to get current location of GPS with id=\"{}\"", gpsId);
+		Gps gps = gpsProvider.getGpsById(gpsId);
+
+		if (!gps.getLastLocation().isPresent()) {
+			LOGGER.error("GPS with id=\"{}\" has no location", gpsId);
+			throw new NoLocationForGpsException(gpsId);
+		}
+
+		LOGGER.info("Successfully retrieved last location for GPS with id=\"{}\"", gpsId);
+
+		return gps.getLastLocation().get();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.tiolktrack.byznass.com/coding_rules#rule_key=squid%3AS3655)